packages/flutter_tools/lib/src/cache.dart

```diff
@@ -105,21 +107,38 @@ class Cache {
     }
   }
 
-  static String _dartSdkVersion;
+  String _dartSdkVersion;
 
-  static String get dartSdkVersion => _dartSdkVersion ??= platform.version;
+  String get dartSdkVersion => _dartSdkVersion ??= platform.version;
 
-  static String _engineRevision;
+  String _engineRevision;
 
-  static String get engineRevision {
-    if (_engineRevision == null) {
-      final File revisionFile = fs.file(fs.path.join(flutterRoot, 'bin', 'internal', 'engine.version'));
-      if (revisionFile.existsSync())
-        _engineRevision = revisionFile.readAsStringSync().trim();
-    }
+  String get engineRevision {
+    _engineRevision ??= getVersionFor('engine');
     return _engineRevision;
   }
 
+  String _engineDartVersion;
+
+  String get engineDartVersion {
+    if (_engineDartVersion == null) {
+      final Directory engineDirectory = getArtifactDirectory('engine');
+      File dartSdkBin;
+      if (platform.isLinux) {
+        dartSdkBin = engineDirectory.childFile('linux-x64/dart-sdk/bin/dart');
+      } else if (platform.isMacOS) {
+        dartSdkBin = engineDirectory.childFile('darwin-x64/dart-sdk/bin/dart');
+      } else if (platform.isWindows) {
+        dartSdkBin = engineDirectory.childFile('windows-x64/dart-sdk/bin/dart.bat');
```

Comment:
I am not sure if this is how it works for Windows.....

Reply:
I don't think it will work (this is shorthand for `engineDirectory.filesystem.file(engineDirectory.fileSystem.path.join(engineDirectory.path, 'windows-x64/dart-sdk/bin/dart.bat'))` -- which is to say, these slashes should be backslashes.

But can you please add a test for `engineDartVersion`, then it'll get run on AppVeyor, and we'll be sure it works :-)